### PR TITLE
Measure grad norm and noise scale in finetuning + fix grad clip in V1 CausalTransformer

### PR DESCRIPTION
**Pull request by nostalgebraist**
- Modifies `CasualTransformer.train` to return gradient norms at the microbatch and single-example level
- Uses these to compute the gradient noise scale estimate `B_simple`, from Appendix A.1 of [An Empirical Model of Large-Batch Training](https://arxiv.org/abs/1812.06162)
- Restores the psum line that was removed [here](https://github.com/kingoflolz/mesh-transformer-jax/commit/02850c46c6b9d957b2a9214c2c6c291269648792#diff-c0219720e7ca7d79c74422022047497e12677c195f846ecf29af21017f6d21dfL21) so V1 grad clipping works properly

(For the psum change, I tried to do something that would turn off the psum in V2, but there's probably a better way to do it.  What I need for this PR to work is just psumming in V1)

----

Computing `B_simple` involves

1. measuring two quantities `G_noise` and `S_noise` that are high-variance across steps
2. taking moving averages to smooth them
3. taking a ratio of the moving averages

I chose to do steps 2-3 inside the script, so we can report the averages and `B_simple` directly to wandb for continual monitoring.  (I don't know a way to compute a ratio of moving averages within wandb itself)

There are some heuristic choices involved, like picking the moving average constant.  One can choose these differently of course -- this code reflects what I currently use, and I've used similar choices in other repos

----

About correctness of the implementation:

I implemented the same thing in mtf gpt-neo earlier, and the results are about the same for similar data / lr.  Both implementations reproduce expected relationships like the roughly inverse one between lr and B_simple.

Longer ago, I calculated B_simple in a (deeply cursed) script for tuning tensorflow gpt-2 on v2-8s, with B_small=20 and B_big=120.  That script gave me larger numbers for B_simple, about 5x.  I'm not sure what explains that difference -- could be a lot of things.

### CODE REVIEW

File: device_train.py [hunk 1]
```diff
@@ -232,14 +240,14 @@ def eval_step(network, data):
 
     val_sets = {}
 
-    for k, v in params['val_set'].items():
-        val_sets[k] = TFRecordNewInputs(f"data/{v}",
-                                        batch_size=(global_val_batch,),
-                                        sample_size=seq)
+    for k, v in params["val_set"].items():
```
Comment (nostalgebraist) [inline device_train.py:243]: Accidental formatting change I think

**Comment by kingoflolz** [comment]
Thanks!
